pathplannerlib/src/sources.patch

```diff
@@ -1,45 +1,56 @@
 --- orig/pathplanner/lib/PathPlanner.cpp	2022-04-03 03:30:25.000000000 -0500
 +++ patched/pathplanner/lib/PathPlanner.cpp	2022-04-03 03:30:37.000000000 -0500
-@@ -16,18 +16,35 @@
+@@ -10,14 +10,52 @@
+ #include <units/velocity.h>
+ #include <vector>
  
- PathPlannerTrajectory PathPlanner::loadPath(std::string name, units::meters_per_second_t maxVel, units::meters_per_second_squared_t maxAccel, bool reversed){
-     std::string line;
-+    wpi::json json;
++#include <wpi/fs.h>
++#ifndef __FRC_ROBORIO__
++#include <robotpy_build.h>
++#include <pybind11/eval.h>
 +
-+    std::vector<std::string> prefixes = {
-+        frc::filesystem::GetDeployDirectory() + "/pathplanner/",
-+        "./",
-+        "/home/lvuser/py/"
-+    };
- 
--    std::string filePath = frc::filesystem::GetDeployDirectory() + "/pathplanner/" + name + ".path";
-+    bool fileFound = false;
-+    std::string errorMessage = "Cannot open file: \n";
++static fs::path getMainPath() {
++    py::gil_scoped_acquire gil;
++    py::dict locals;
++    py::exec(R"(
++        import sys, os.path
++        main = sys.modules['__main__'];
++        if hasattr(main, '__file__'):
++            main_path = os.path.abspath(os.path.dirname(main.__file__))
++
++    )", py::globals(), locals);
++
++    if (locals.contains("main_path")) {
++        return fs::path(py::cast<std::string>(locals["main_path"]));
++    } else {
++        return fs::current_path();
++    }
++}
++
++#endif
++
+ using namespace pathplanner;
  
--    std::error_code error_code;
--    wpi::raw_fd_istream input{filePath, error_code};
-+    for(auto prefix : prefixes){
-+        std::string filePath = prefix + name + ".path";
+ double PathPlanner::resolution = 0.004;
  
--    if(error_code){
--        throw std::runtime_error(("Cannot open file: " + filePath));
-+        std::error_code error_code;
-+        wpi::raw_fd_istream input{filePath, error_code};
-+
-+        if(!error_code){
-+            fileFound = true;
-+            input >> json;
-+            break;
-+        }
-+
-+        errorMessage += "  - " + filePath + "\n";
-     }
++
++static fs::path getDeployDirectory() {
++    static fs::path deployPath =
++#ifdef __FRC_ROBORIO__
++        fs::path(frc::filesystem::GetOperatingDirectory()) / "py" / "deploy" / "pathplanner";
++#else
++        getMainPath() / "deploy" / "pathplanner";
++#endif
++    return deployPath;
++}
```

Comment:
This, without the "pathplanner", should go in  robotpy-wpilib. It has the filesystem functions disabled.

Reply:
I suppose that's true.